### PR TITLE
add suffix to article slug in case of slug duplication

### PR DESCRIPTION
**Pull request by ahilles107**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | yes
| New feature?              | no
| BC breaks?                | no
| Deprecations?             | no
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | 
| License                   | AGPLv3

### CODE REVIEW

File: features/packages/send_package_with_this_same_slug_twice.feature [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    """
+    Then the response status code should be 201
+    And I am authenticated as "test.user"
+    And I add "Content-Type" header equal to "application/json"
+    Then I send a "GET" request to "/api/v1/content/articles/abstract-html-test"
+    Then the response status code should be 200
+    And I am authenticated as "test.user"
+    And I add "Content-Type" header equal to "application/json"
+    Then I send a "GET" request to "/api/v1/content/articles/abstract-html-test-492c31d1"
+    Then the response status code should be 200
```
Comment (takeit) [inline features/packages/send_package_with_this_same_slug_twice.feature]: new line is missing

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ This changelog references the relevant changes (bug and security fixes) done in
 To get the diff for a specific change, go to https://github.com/superdesk/web-publisher/commit/XXX where XXX is the change hash
 
 * 1.4.0
+ * improvement [#560] Add prefix (generated from package guid) for duplicated article slugs
```
Comment (takeit) [inline CHANGELOG.md]: it should be `suffix`

File: src/SWP/Bundle/CoreBundle/EventListener/DuplicatedSlugListener.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        /** @var ArticleInterface $article */
+        $article = $event->getArticle();
+
+        $existingArticle = $this->articleRepository->findOneBy(['slug' => $article->getSlug()]);
+        if (null === $existingArticle) {
+            return;
+        }
+
+        $hash = substr(md5($article->getPackage()->getGuid()), 0, 8);
```
Comment (takeit) [inline src/SWP/Bundle/CoreBundle/EventListener/DuplicatedSlugListener.php]: you can use `RandomStringGenerator` for that suffix generation

Comment (ahilles107) [reply src/SWP/Bundle/CoreBundle/EventListener/DuplicatedSlugListener.php]: We don't want it to be random. This way we can predict that hash based on package guid (in tests, or in future cases).

**Pull request merged**